pkg/storage/ingest/pusher.go

```diff
@@ -241,163 +252,281 @@ func (c pusherConsumer) shouldLogClientError(ctx context.Context, err error) (bo
 	return optional.ShouldLog(ctx)
 }
 
-type multiTenantPusher struct {
+// sequentialStoragePusher receives mimirpb.WriteRequest which are then pushed to the storage one by one.
+type sequentialStoragePusher struct {
+	metrics *pusherConsumerMetrics
+
+	pusher Pusher
+}
+
+// newSequentialStoragePusher creates a new sequentialStoragePusher instance.
+func newSequentialStoragePusher(metrics *pusherConsumerMetrics, pusher Pusher) sequentialStoragePusher {
+	return sequentialStoragePusher{
+		metrics: metrics,
+		pusher:  pusher,
+	}
+}
+
+// PushToStorage implements the PusherCloser interface.
+func (ssp sequentialStoragePusher) PushToStorage(ctx context.Context, wr *mimirpb.WriteRequest) error {
+	// TODO: What about time??
```

Comment:
what about it? :smile: did you mean to address this now or before we merge to `main`?

Reply:
I don't have any telemetry here so I hope you don't mind if the comment stays, I'm trying to avoid adding any until we're set this is the direction we want to take.